elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -187,35 +202,43 @@ default void updateToOneRelation(DataStoreTransaction relationTx,
                                      Object entity,
                                      String relationName,
                                      Object relationshipValue,
-                                     RequestScope scope) { };
+                                     RequestScope scope) { }
 
     /**
+     * Get an attribute from an object.
+     *
      * @param entity - The object which owns the attribute.
      * @param attributeName - name of the attribute.
      * @param scope - contains request level metadata.
      */
-    default Object getAttribute(
-            Object entity,
-            String attributeName,
-            RequestScope scope) {
+    default Object getAttribute(Object entity,
+                               String attributeName,
+                               RequestScope scope) {
         com.yahoo.elide.core.RequestScope requestScope;
         try {
             requestScope  = (com.yahoo.elide.core.RequestScope) scope;
         } catch (ClassCastException e) {
             throw new ClassCastException("Fail trying to cast requestscope");
         }
+
+        Object val = PersistentResource.getValue(entity, attributeName, requestScope.getDictionary());
+        return val;
+
     }
 
     /**
+     * Set an attribute on an object in the data store.
+     *
+     * Elide core will update the in memory representation of the objects to the requested state.
+     * This function allow a data store to optionally persist the attribute if needed.
+     *
      * @param entity - The object which owns the attribute.
      * @param attributeName - name of the attribute.
      * @param attributeValue - the desired attribute value.
      * @param scope - contains request level metadata.
      */
-    default void setAttribute(
-            Object entity,
-            String attributeName,
-            Object attributeValue,
-            RequestScope scope) {
-    }
+    default void setAttribute(Object entity,
+                      String attributeName,
+                      Object attributeValue,
+                      RequestScope scope) { }
```

Comment:
Interesting question and I think the answer depends on cleanliness of code and separation of concern.

The way this is done right now, `PersistentResource` actually sets the in-memory value of the object. Doing it again here would be redundant. In the case of `InMemory` or `Hibernate` proxies, a noop here is effective.